test/unit/org/apache/cassandra/config/DatabaseDescriptorTest.java

```diff
@@ -274,55 +274,56 @@ public void testTokensFromString()
         assertTrue(DatabaseDescriptor.tokensFromString(null).isEmpty());
         Collection<String> tokens = DatabaseDescriptor.tokensFromString(" a,b ,c , d, f,g,h");
         assertEquals(7, tokens.size());
-        assertTrue(tokens.containsAll(Arrays.asList(new String[]{"a", "b", "c", "d", "f", "g", "h"})));
+        assertTrue(tokens.containsAll(Arrays.asList("a", "b", "c", "d", "f", "g", "h")));
     }
 
     @Test
-    public void testExceptionsForInvalidConfigValues() {
+    public void testExceptionsForInvalidConfigValues()
+    {
         try
         {
             DatabaseDescriptor.setColumnIndexCacheSize(-1);
-            fail("Should have received a ConfigurationException column_index_cache_size_in_kb = -1");
+            fail("Should have received a IllegalArgumentException column_index_cache_size = -1");
         }
-        catch (ConfigurationException ignored) { }
+        catch (IllegalArgumentException ignored) { }
```

Comment:
I'd consider catching `IllegalArgumentException` and wrapping it in a `ConfigurationException` where possible to avoid changing the contract of the public JMX bits that sort of assume that. I'm not sure if there's actually any tooling that would depend on it, but perhaps better to be safe?

Reply:
Changed the `XSpec` classes to throw `ConfigurationExceptions` instead as per our conversation on Slack.